src/models/neural_networks/base.py

```diff
@@ -397,4 +406,20 @@ def get_morris_gradient(self, x: TrainData) -> TrainData:
             else:
                 output_dict[key] = None
 
+        if save_explanations:
```

Comment:
is this in the original `TrainData` dimensions or is it a `latlon` grid like the original xarray data?

Reply:
or have i misinterpreted how these gradients are calculated? Are they not also per pixel, per variable?

Reply:
so this is the same as the shap values, in a numpy array. This means the `all_shap_for_file` function can be used (https://github.com/esowc/ml_drought/blob/master/src/analysis/plot_shap.py) to turn the outputs into a netcdf file.

Reply:
the only issue i see with working with those `plot_shap` functions is the hardcoding of the months. Our data will very likely become daily / weekly at some point and we want the flexibility of working with hourly data too (e.g. if emulating ERA5 hourly). That's why i think writing conversions to xarray objects makes sense because we always have the timesteps/timestamps